mitodl/mit-open
# First iteration of field pages

### What are the relevant tickets?
https://github.com/mitodl/mit-open/issues/602

### Description (What does it do?)
This PR moves field functionality from open discussions to the new open UI. It does not include list functionality since we probably don't need it. 

### Screenshots (if appropriate):
<img width="1425" alt="Screenshot 2024-03-18 at 11 22 36 PM" src="https://github.com/mitodl/mit-open/assets/1934992/8e5f5ea9-511c-44b3-8b15-04b640013f11">
<img width="1419" alt="Screenshot 2024-03-18 at 11 23 03 PM" src="https://github.com/mitodl/mit-open/assets/1934992/a141061c-a204-443a-9da3-10e4b0ccfb1d">

<img width="1429" alt="Screenshot 2024-03-18 at 11 23 16 PM" src="https://github.com/mitodl/mit-open/assets/1934992/60a46e8d-8625-4d01-9bb7-b06b9950db1b">

### How can this be tested?
- If you don't have a field locally already, create one at http://localhost:8063/api/v1/fields/ 
- Go to http://localhost:8063/fields/<your field>/ you should see your field's name and description
- Go to http://localhost:8063/fields/<your field>/manage - if you are not logged in, you should see a message that you don't have permission to access the page
- Login and go back to  http://localhost:8063/fields/<your field>/. Click the gear at the right hand side of the banner to go to the field management page
-  You should be able to change the title and description in the "appearance" tab
- You should be able to add new widgets 

### Additional Context
There will be follow up tickets to clean up the documentation and api call structure for the field and widget_list api calls (https://github.com/mitodl/mit-open/issues/643 and https://github.com/mitodl/mit-open/issues/644)
